### PR TITLE
Attempt to make typography approach more standard and predictable

### PR DESCRIPTION
**Pull request by ashley-hebler**
#### What's this PR do?
Renames a bunch of typography classes

##### Classes added (if any)
- t-links
- t-links-underlined
- t-links-underlined-hover
- t-lsp-{size} letter-spacing
- t-lh-{size} line-height
- t-{weight} font-weight

##### Classes removed (if any)
- t-linkstyle
- t-linkstyle--underlined
- t-linkstyle--underlined-subtle
- t-linkstyle--blue (never used)
- t-prose
- t-smallcaps
- t-space-{size} (replaced with `t-lh-{size}`)
- t-uppercase--wide/extra-wide (replaced with `t-lsp-{size}`)
- t-widgethead (never used)

#### Why are we doing this? How does it help us?
I created some overlap within a ton of our typography classes. For example, it was hard to know when to use `t-smallcaps` vs `t-uppercase--extra-wide` etc. There was also some naming convention standardization that was desperately needed.

#### How should this be manually tested?
`yarn dev`

See: [typography](http://localhost:3000/pages/typography/index.html)

The key for testing on this one is making sure the naming is intuitive. Does anything throw you off or give you pause?

#### Does this introduce a breaking change where queso-ui is used in the wild? If so, is there a relevant branch/PR to accompany this release?

This one is pretty disruptive and will require two thoughtful /donations and /texastribune PRs. I'm creating a list below of the procedure for those.

##### Switch ups needed in other repos

t-byline
- t-lsp-m
- t-uppercase
- remove t-smallcaps and t-smallcaps--light

t-headline
- remove `--article` modifier
- t-lh-s
- t-serif

t-links (t-linkstyle)
- Now a helper
- t-linkstyle -> t-links
- t-linkstyle t-linkstyle--uppercase -> t-links-underlined

t-prose
- May need t-serif and t-links-underlined

t-sectionhead
- t-weight-bold (or if it's a heading, it'll just inherit)
- t-lsp-b

t-smallcaps
- Removed completely
- t-uppercase
- t-lsp-b
- t-weight-bold

t-smallcaps--light
- Removed completely
- t-uppercase
- t-lsp-m
- t-weight-normal

t-uppercase--wide
- t-lsp-b

t-uppercase--extra-wide
- t-lsp-m

t-space-base
- t-lh-b

t-space-heading-s
- t-lh-s

t-space-heading-m
- t-lh-m

t-space-heading-l
- t-lh-l

#### TODOs / next steps:

* [ ] *Account for the above list and ensure no visual changes occur*

### CODE REVIEW

File: assets/scss/5-typography/_headline.scss [hunk 1]
```diff
@@ -13,12 +13,9 @@
 .t-headline {
   @include font-setting('secondary');
   @include underlined-link-parent('true');
```
Comment (ashley-hebler) [inline assets/scss/5-typography/_headline.scss]: I know we have helpers for these, but `t-headline` feels like it should contain some presets just for ease of use.

**Comment by ashley-hebler** [comment]
I'm at an impasse with this. I can't decide if certain typography helpers should have some baked-in rules (regardless of the fact that helpers like line-height and letter-spacing _could_ achieve those.
More specifically for classes like `t-headline`, `t-byline` and `t-sectionhead`
- Should t-headline include a font-fam or should you just always plan to use it with `t-serif`?
- Should t-byline always have `t-uppercase` and `t-lsp-m` when it's used or should you get those rules with that class to standardize that style?

I like the "tack on helper classes based on the documentation's example approach" (especially with font-size), but it's hard to know where to draw that line. Maybe it's easier if there isn't one and we stay true to the mix 'n match approach. Where do you stand on this @AndrewGibson27 and @emilyyount?

**Comment by emilyyount** [comment]
@ashley-hebler: Thank you for thinking this through. I agree it's tricky. Happy to discuss in person with you and Andrew, but here are some initial thoughts ..

- PROS: This change adds a lot of flexibility and probably eliminates the need to do a lot of overwriting of classes like t-byline. And it makes it easier to apply the look of t-byline to something that isn't a byline. This could also really help data visuals incorporate queso without having to overwrite much. I imagine they have a slightly different headline style for their template and this would allow them to use queso and still vary things where needed, just by adding a class instead of overwriting in their template. Overwriting in their template could be cumbersome longer term as queso evolves.
- CONS: I could see this added flexibility translating to less consistency around the website — for example, prose not quite being the same across templates — but that feels like something we can be intentional about. One solution could be to have examples in the documentation for common type class "recipes".

**Comment by ashley-hebler** [comment]
@emilyyount, yes I keep coming back to that idea of recipes. I'll think more on that, but that's good feedback. That's a great point on how these changes would interact in a data-viz world.

**Comment by AndrewGibson27** [comment]
I favor not baking in rules unless absolutely necessary. In an ideal world, everything would look the same everywhere -- and in that world, we would bake. But we don't live in that world. There has been no shortage of times where art/others have, for example, asked for our standard headline style _minus one little thing_. And that then created lots of overriding and thus lots of messes. So I favor being explicit, even if it more verbose.

I'm all good with this, though! I'm sure the HTML updating will be a bit tedious, but I have full faith you will nail it! LMK if you need advising on anything.

**Comment by ashley-hebler** [comment]
Sounds great! That makes sense to me too, @AndrewGibson27. I'll use tomorrow to release this and start crafting those PRs in the other two repos.

**Pull request merged**